solcore/absorption_calculator/transfer_matrix.py

```diff
@@ -4,14 +4,31 @@
 """
 import numpy as np
 import solcore
-import tmm as old_tmm
 from solcore.interpolate import interp1d
 from solcore.structure import ToStructure
 from solcore.absorption_calculator import tmm_core_vec as tmm
+from functools import lru_cache, wraps
 
 degree = np.pi / 180
 
 
+def np_cache(function):
```

Comment:
Please add a doctoring explaining what this (rather obscure) function does, as well as including the link to the StackOverflow answer you got it from: https://stackoverflow.com/a/52332109/3778792